crates/ruff/src/rules/ruff/rules/implicit_optional.rs

```diff
@@ -180,60 +179,66 @@ impl<'a> TypingTarget<'a> {
                 ..
             }) => Some(TypingTarget::None),
             Expr::Constant(ast::ExprConstant {
-                value: Constant::Str(_),
+                value: Constant::Str(string),
+                range,
                 ..
-            }) => Some(TypingTarget::ForwardReference),
-            _ => semantic.resolve_call_path(expr).and_then(|call_path| {
-                if semantic.match_typing_call_path(&call_path, "Any") {
-                    Some(TypingTarget::Any)
-                } else if matches!(call_path.as_slice(), ["" | "builtins", "object"]) {
-                    Some(TypingTarget::Object)
-                } else {
-                    None
-                }
-            }),
+            }) => parse_type_annotation(string, *range, checker.locator)
+                // In case of a parse error, we return `None` to avoid false positives.
+                .map_or(Some(TypingTarget::None), |(expr, _)| {
```

Comment:
Maybe... `Any`? Or `Object`? Or a special `Unknown` type?

Reply:
Oh, good point. `Any` or `Object` would make more sense. Adding a new type would require handling it elsewhere so I'm leaning away from it.

Reply:
Trust you, whichever you prefer.